ts/screens/preferences/ServicesScreen.tsx

```diff
@@ -50,6 +53,22 @@ class ServicesHomeScreen extends React.Component<Props> {
     });
   };
 
+  public componentWillMount() {
+    this.checkForExperimentalFeatures();
+  }
+
+  private handleWillFocus = () => {
+    this.checkForExperimentalFeatures();
```

Comment:
You are creating a loop here.
1) You come here with experimentalFeatures flag enabled.
2) You will be redirected to the NEW ServiceHome
3) Than an Android you click the hardware back you will return to the old ServiceHome (you are not removing the screen from the history) but this method will bring you again to the new ServiceHome.